src/compiler/pre-transform/index.test.ts

```diff
@@ -0,0 +1,221 @@
+
+import { codemodLegacyNodes } from './index';
+
+import { getSvelteAST } from '#parser/ast';
+
+describe(codemodLegacyNodes.name, () => {
+  it("replaces 'export const meta' with 'defineMeta'", async ({ expect }) => {
+    const code = dedent(`
+      <script context="module">
+        import { Story } from "${pkg.name}";
+
+        /** This is a description for the **Button** component stories. */
+				export const meta = {
+          title: "Atoms/Button",
+          component: Button,
+        };
+      </script>
+
+			<!-- This is a description for **default** Button. -->
+      <Story name="Default" />
+    `);
+    const ast = getSvelteAST({ code });
+    const transformed = await codemodLegacyNodes({ ast });
+
+    expect(print(transformed)).toMatchInlineSnapshot(`
+      "<script context="module">
+      	import { defineMeta } from "@storybook/addon-svelte-csf";
+
+      	/** This is a description for the **Button** component stories. */
+      	const { Story } = defineMeta({ title: "Atoms/Button", component: Button });
+      </script>
+
+      <!-- This is a description for **default** Button. -->
+      <Story name="Default" />"
+    `);
+  });
+
+  it("replaces 'Meta' with inserting 'defineMeta' to module tag", async ({ expect }) => {
+    const code = dedent(`
+      <script context="module">
+        import { Meta, Template } from "${pkg.name}";
+      </script>
+
+			<!-- This is a description for the **Button** component stories. -->
+      <Meta title="Atoms/Button" component={Button} />
+    `);
+    const ast = getSvelteAST({ code });
+    const transformed = await codemodLegacyNodes({ ast });
+
+    expect(print(transformed)).toMatchInlineSnapshot(
+      `
+      "<script context="module">
+      	import { defineMeta } from "@storybook/addon-svelte-csf";
+
+      	/** This is a description for the **Button** component stories. */
+      	const { Story } = defineMeta({ title: "Atoms/Button", component: Button });
+      </script>"
+    `
+    );
+  });
+
+  it("replaces 'Template' with snippet block", async ({ expect }) => {
+    const code = dedent(`
+      <script context="module">
+        import { Meta, Template } from "${pkg.name}";
+      </script>
+
+			<!-- This is a description for the **Button** component stories. -->
+      <Meta title="Atoms/Button" component={Button} />
+
+      <Template let:args>
+        <Button {...args} />
+      </Template>
+
+      <Story name="Default" />
+    `);
+    const ast = getSvelteAST({ code });
+    const transformed = await codemodLegacyNodes({ ast });
+
+    expect(print(transformed)).toMatchInlineSnapshot(`
+			"<script context="module">
+				import { defineMeta } from "@storybook/addon-svelte-csf";
+
+				/** This is a description for the **Button** component stories. */
+				const { Story } = defineMeta({ title: "Atoms/Button", component: Button });
+			</script>
+
+			{#snippet children(args)}
+				<Button {...args} />
+			{/snippet}
+			<Story name="Default" />"
```

Comment:
this looks incorrect? `children` isn't referenced

Reply:
You're right, is totally incorrect. And I'm puzzled on what to do with this case.

Reminder:
1. Component `<Template>` attribute `id` is optional
2. `<Story />` component(s) by default were using the `default` template _(if existed)_, but the reference isn't visible on the stories file _(is a runtime mechanism)_

In this case, is easy to assume that the best solution to this problem is to achieve the following result:

```svelte
<script context="module">
import { defineMeta, setTemplate } from "@storybook/addon-svelte-csf";
/** This is a description for the **Button** component stories. */
const { Story } = defineMeta({ title: "Atoms/Button", component: Button });
</script>

<script>
setTemplate(template);
</script>

{#snippet template(args)}
   <Button {...args} />
{/snippet}

<Story name="Default" />

<Story name="Other" />
```

To create a codemod for this one complicates the logic a bit because:
1. I need to create a `<script>` instance
2. I need to add `setTemplate` to the import declaration for this addon inside the `<script module>` tag.

Do you see a simpler approach to this case?

Reply:
1. When you find a `<Template/>` without an explicit id, convert it to a snippet with a namespaced name so it won't collide with others, like `{#snippet sb_default_template_1()}`. Also give it a number, in the case where multiple id-less Templates are found, they can have number 1, 2 3... I don't know why anyone would have multiple id-less Templates, but just to be on the safe side?
2. When you find a `<Story/>` without an explicit template reference, make it reference the first default, eg. `children={sb_default_template_1}`

Then you don't need the `setTemplate` complexity at all?

Reply:
I did give it a try. cfc7beb82f63157e791470f73cfb1a5efd4e3c8b

Had to refactor a little bit by passing down the state from `zimmerframe` walk down to transformation function.

I only found one gotcha in `2.` point. Assuming someone did create a bizarre stories _(invalid, but AFAIK has a fallback mechanism behaviour)_, like this:

```svelte
      <Template let:context>
        <p>{context.args}</p>
      </Template>

      <Story name="Default" />

      <Template let:args>
        <Button {...args} />
      </Template>

      <Story name="NextOne" />
```

... which children should the second story `NextOne` refer to? I assumed it should match the latest one. And the tests shows current ouput:

```svelte
      {#snippet sb_default_template_1(_args, context)}
      	<p>{context.args}</p>
      {/snippet}

      <Story name="Default" children={sb_default_template_1} />

      {#snippet sb_default_template_2(args)}
      	<Button {...args} />
      {/snippet}

      <Story name="NextOne" children={sb_default_template_2} />
```

Hope that's ok. Regardless, this case is very edgy, hope it never happens. It was a little eye-opening, because I see why sometimes there's a need to be more conservative with API design. The previous API design is... too *flexible* 😅

Reply:
Argh, there's an edge case that comes with the currently implemented solution. I had to change the components order: c2b6b827913a

Explanation: CI was failing, because `<Story name="default" />` were transformed into `<Story name="default" children={sb_default_template_2} />`. And the reference `sb_default_template_2` isn't defined because the second `<Template>` has an `id` which is used as an identifier for its own transformation to snippet.

I could probably filter out '*id-less*' templates, so this bizarre edge case doesn't happen. But I'll put it on hold until I get to know your opinion on this solution.

Reply:
Alright, I tested out what the behavior in v4 was - because we should just mimic that. With the follow stories:

```svelte
<Template let:args>
	<Button {...args}>
		First template
	</Button>
</Template>

<Story name="First Story" args={{ rounded: true }} />

<Template let:args>
	<Button {...args}>
		Second template
	</Button>
</Template>

<Story name="Second Story" source args={{ rounded: false }} />
```

You get these two, super weird stories:

![image](https://github.com/user-attachments/assets/3540ff0a-061e-447d-8c45-b8488f738904)

![image](https://github.com/user-attachments/assets/bcedb8e6-b7e8-4cf3-9ae5-21e5b402b45f)

So it seems that if you have multiple templates without `id`, they just combine into a single render?! I cannot fathom why users would ever want this explicitly, and they'd also notice this right away because the whole story renders double. So there's 0.0001 % chance this will ever happen. The same thing happens btw if multiple Templates have the same id.

We don't need to support multiple templates without id/with the same id. Ideally we'd throw a nice error if we detect this, but if that's too hard, then don't even waste time on it, just let it fail however it does naturally.

Reply:
Huh. It might be some issue with logic on the runtime scope of this addon.

Regardless, I see a way to throw an error if there's more than two *'id-less'* `<Template>` components and it certainly feels like an easier solution for this edge scenario.

Reply:
b1bf0c3b665e